components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -10078,8 +10078,8 @@ public boolean isScopeKeyAssignedLocally(APIIdentifier apiIdentifier, String sco
                     String apiName = rs.getString("API_NAME");
                     // Check if the provider name and api name is same.
                     // Return false if we're attaching the scope to another version of the API.
-                    return !(provider.equals(APIUtil.replaceEmailDomainBack(apiIdentifier.getProviderName()))
-                            && apiName.equals(apiIdentifier.getApiName()));
+                    return !(provider.equals(APIUtil.replaceEmailDomainBack(identifier.getProviderName()))
```

Comment:
Do we need to do this anymore? we could check the returned UUID is matching the passing uuid